applications/plugins/SofaPython/python/SofaPython/Quaternion.py

```diff
@@ -35,6 +35,10 @@ def angle(q):
     """get angle in rad"""
     return 2.0* math.acos(re(q))
 
+def product(qa, qb):
```

Comment:
Hi, 
I think your "product" function does exactly the same as the "prod" function just below (I checked). 
I prefer yours, because of the comment and the fact that the formula is easier to understand/verify.
So maybe we should remove the "prod" function?

Reply:
Hi,
Python noob here, I see `hstack` and `numpy`, I think performance issue. @matthieu-nesme could you help?

Reply:
If it is the same, why adding a new function?
Rather than removing the `prod` function (that is used in existing code), I would keep it, but replace its code with the new implem (personnaly I would keep the old implementation as a comment as I find it clearer/closer to the math).

But please be sure that is *exactly* the same.

Also, rather than using directly the [] operator, I would use re(a), im(b)[0], im(b)[1], im(b)[2]. That way it is easier to read, and independent to the scalar order.

Reply:
There is a performance issue in both cases:
- `hstack` + list (!) is useless and can be replaced by `res = np.zeros(4); res[:3] = ...; res[-1] = ...;`
- yet the proposed version is probably even less efficient since it does all the computations in python which is *waaay* slower than letting numpy doing its stuff in C